reframed/replay.py

```diff
@@ -53,7 +62,96 @@ def __fix_types(self, data):
 		data["mappinginfo"]["fighterstatus"]["specific"] = {int(k): v for k, v in data["mappinginfo"]["fighterstatus"]["specific"].items()}
 		data["mappinginfo"]["hitstatus"] = {int(k): v for k, v in data["mappinginfo"]["hitstatus"].items()}
 		data["mappinginfo"]["stageid"] = {int(k): v for k, v in data["mappinginfo"]["stageid"].items()}
-		
+
+	def __unpack_modern_v1_0(self, filename):
+		data = open(filename, "rb").read()
+		# Starts off with "RFR1" then has number of entries in table.
+		num_entries_in_table = int.from_bytes(data[4:5], "little")
+		# Flag, like ("META", "MAPI", "FDAT", "VIDM", or "VIDE") to (offset, size) tuple.
+		flag_to_bytes_offset = {}
+		self.__data = {}
+		for i in range(num_entries_in_table):
+			start = 5 + (i * 12)
+			flag = data[start : start + 4].decode("utf-8")
+			offset = int.from_bytes(data[start + 4 : start + 8], "little")
+			size = int.from_bytes(data[start + 8 : start + 12], "little")
+			flag_to_bytes_offset[flag] = (offset, size)
+
+		# TODO: "VIDE"
+		for (flag, key) in [("MAPI", 'mappinginfo'), ("META", None), ("VIDM", 'videoinfo')]:
+			# This is a typical json.
+			offset, size = flag_to_bytes_offset[flag]
+			json_bytes = data[offset : offset + size]
+			if key:
+				self.__data[key] = json.loads(json_bytes)
+			else:
+				self.__data.update(json.loads(json_bytes))
+
+		# Frame Data
+		if "FDAT" in flag_to_bytes_offset:
+			# This is a typical json.
+			offset, size = flag_to_bytes_offset["FDAT"]
+			frame_data_bytes = data[offset : offset + size]
+
+			# major_version = int.from_bytes(frame_data_bytes[0:1], "little")
+			# minor_version = int.from_bytes(frame_data_bytes[1:2], "little")
+			# uncompressed_size = int.from_bytes(frame_data_bytes[2:6], "little")
+
+			# The data was compressed using zlib. Now decompress it.
+			decompressed_frame_data_bytes = zlib.decompress(frame_data_bytes[6:])
+
+			num_frames = int.from_bytes(decompressed_frame_data_bytes[0:4], "little")
+			num_players = int.from_bytes(decompressed_frame_data_bytes[4:5], "little")
+
+			offset = 5
+
+			self.__data['playerstates'] = {}
+			for p in range(num_players):
+				if p not in self.__data['playerstates']:
+					self.__data['playerstates'][p] = []
+
+				# < = little endian
+				# I = unsigned int
+				for i in range(num_frames):
+					(
+						timestamp,
+						frames_left,
```

Comment:
I believe I'm doing something wrong with `frames_left`.  For the example rfr file I'm getting.:

```
...
frames_left:  9642
frames_left:  9642
frames_left:  9642
frames_left:  9642
frames_left:  9642
frames_left:  9642
frames_left:  25199
frames_left:  25198
frames_left:  25197
frames_left:  25196
frames_left:  25195
...
```